pkg/cmd/cli/describe/chaindescriber.go

```diff
@@ -189,3 +198,68 @@ func outputHelper(info, namespace string, singleNamespace bool) string {
 	}
 	return fmt.Sprintf("<%s %s>", namespace, info)
 }
+
+type byID []graph.Node
+
+func (a byID) Len() int {
+	return len(a)
+}
+
+func (a byID) Less(i, j int) bool {
+	return a[i].ID() < a[j].ID()
+}
+
+func (a byID) Swap(i, j int) {
+	t := a[i]
+	a[i] = a[j]
+	a[j] = t
+}
+
+// DepthFirst implements stateful depth-first graph traversal.
+// Modifies behavior of visitor.DepthFirst to allow nodes to be visited multiple
+// times as long as they're not in the current stack
+type DepthFirst struct {
```

Comment:
> @deads2k should we move this under the graph api?

It's an unusual traversal somewhere between DFS and all DFS paths.  I'm fine with leaving it here.